### DISCUSSION TITLE
error: Cannot read properties of undefined (reading 'data')

### DISCUSSION THREAD

**Discussion by RomanBON** [Four-Lights-NL/strapi-plugin-deep-copy#53] [Q&A]
Hello! Error after clicking "Create clone" button.
error: Cannot read properties of undefined (reading 'data')
TypeError: Cannot read properties of undefined (reading 'data')
    at /Users/me/cms/node_modules/@fourlights/strapi-plugin-deep-copy/dist/server/utils/prepareForCopy.js:42:45
![Screenshot 2023-04-28 at 13 16 52](https://user-images.githubusercontent.com/11601489/235133904-ae612e53-d7f8-4d62-9cf3-b22c07c818b1.png)

**Comment by trijpstra-fourlights** [comment]
I've published a new version which should return a more helpful error (and fixes rollback behavior when an error has occurred).

Perhaps you can try it out and see if the returned error makes more sense to you.

In my experience, most errors boil down to:

* failed validation of the newly created entities (e.g. an enumeration has changed in values, but the object your copying still has the old, now incorrect, value)
* unique fields not being unique (you can try to do use a different internal id)

Other than that, there's a fair chance that you're encountering a new situation that isn't handled yet in the plugin. As described in the README, the plugin is still in development and shouldn't be used in production unless you've verified that it's working as expected for your specific data model!

If you've tried the new version, could you update your issue here? Thanks!
